library/Fission/DNS.hs

```diff
@@ -0,0 +1,34 @@
+module Fission.DNS (splitRecord) where
+
+import           Fission.Prelude
+
+import           RIO.NonEmpty
+import qualified RIO.Text as Text
+import qualified RIO.List as List
+
+
+-- if less than 256 bytes, return normal. 
+-- if larger then split into chunks & prefix with a 3-char decimal prefix  & semicolon delimiter ("001;")
+splitRecord :: Text -> NonEmpty Text
+splitRecord txt = 
+  if Text.length txt <= 256
+    then 
+      pure txt
+
+    else 
+      Text.chunksOf 252 txt
+      |> List.zip [0..]
+      |> (joinRecord <$>)
+      |> nonEmpty
+      |> maybe (pure "") identity
+
+joinRecord :: (Integer, Text) -> Text
+joinRecord (index, txt) =  
+  textShow index
+  |> padToLen 3 '0'
+  |> (<> ";" <> txt)
```

Comment:
Also would loose the pipes here, especially with the `<>`s

```haskell
padToLen 3 ‘0’ (textDisplay index) <> “;” <> txt
```

Reply:
You may actually be able to get rid of `padToLen` with some laziness:

```haskell
Text.takeEnd 3 (Text.replicate 2 “0” <> textDisplay index) <> ”;” <> txt
```

Reply:
As mentioned elsewhere, `Natural` would also be good here. We don’t have negative indexes.

Also, I’m using `textDisplay` instead of `textShow` since that’s actually built-in to RIO, and likely is very efficient.

Reply:
> Text.takeEnd 3 (Text.replicate 2 “0” <> textDisplay index) <> ”;” <> txt
oh yeah this is slick